client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -110,7 +113,8 @@
     }
     
     private EventRead<Type> readNextEventInternal(long timeout) throws ReaderNotInReaderGroupException, TruncatedDataException {
-        long waitTime = Math.min(timeout, ReaderGroupStateManager.TIME_UNIT.toMillis());
+        long waitTime = Math.min(timeout, Math.min(ReaderGroupStateManager.TIME_UNIT.toMillis(),
+                BASE_READER_WAITING_TIME_MS + groupState.getNumberOfReaders()));
```

Comment:
Thank you @RaulGracia .
Observing that we have now changed the logic for wait time from 
`READER_WAITING_TIME_MS * numReaders` to `BASE_READER_WAITING_TIME_MS  + numReader` i.e. the wait time increases by a 1ms per new reader added.
I have a couple of question regarding this
- How does the CPU utilization and latency numbers vary between using `BASE_READER_WAITING_TIME_MS  + numReader` v/s a constant value ?
- The perf numbers in the PR description are based on the previous scheme right?

Reply:
Thanks @shrids, yes, you are right, let me reply to your comments:
- I have executed Pravega Benchmark in my laptop (not in a k8 pod) with 1 producer (writing 1 e/s) and 50 consumers (50-segment stream). With the current approach, the CPU utilization of the process (that includes the writer and the read) is generally below 2-3%. Note that in this idleness situation is where the timeout defined is exercised the most.
- Based on your question, I tried the constant value. For the same experiment as executed before, I see that the CPU consumption for 50 readers case is similar to the previous case. The only differential factor is that during the creation of the reader group it takes high CPU consumption 90%-95% for a longer time compared to the previous case (in which this also happens, but for shorter period of time).
- Yes, the numbers are based on the previous scheme. Current numbers are the same for 1 reader and better as the number of reader increase (as the timeout grows less than before based on the number of readers).

Based on that, and given that in real cases multiple readers will be distributed across multiple machines, I will simplify the PR and leave a constant value. This will improve performance for low event rates until we find a safe async way to do the same.